cristin-organization/src/test/java/no/unit/nva/cristin/organization/FetchCristinOrganizationTest.java

```diff
@@ -0,0 +1,192 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.zalando.problem.Problem;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static com.google.common.net.HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static no.unit.nva.cristin.organization.FetchCristinOrganizationHandler.IDENTIFIER_PATTERN;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID;
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static nva.commons.apigateway.ApiGatewayHandler.ALLOWED_ORIGIN_ENV;
+import static nva.commons.apigateway.ApiGatewayHandler.MESSAGE_FOR_RUNTIME_EXCEPTIONS_HIDING_IMPLEMENTATION_DETAILS_TO_API_CLIENTS;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+class FetchCristinOrganizationTest {
+
+    public static final ObjectMapper restApiMapper = JsonUtils.dtoObjectMapper;
+    public static final String IDENTIFIER = "identifier";
+    public static final String IDENTIFIER_VALUE = "1.0.0.0";
+    public static final String ORGANIZATION_NOT_FOUND_MESSAGE = "Organization not found: ";
+    private static final String IDENTIFIER_NULL_ERROR = "Identifier is not a valid Organization identifier: null";
+    FetchCristinOrganizationHandler fetchCristinOrganizationHandler;
+    private CristinApiClient cristinApiClient;
+    private Environment environment;
+    private ByteArrayOutputStream output;
+    private Context context;
+
+    @BeforeEach
+    public void setUp() {
+
+        environment = mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn("*");
+        context = mock(Context.class);
+        cristinApiClient = new CristinApiClient();
+        output = new ByteArrayOutputStream();
+        fetchCristinOrganizationHandler = new FetchCristinOrganizationHandler(cristinApiClient, environment);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "1.0.0.0, true",
+            "1.0.0, false",
+            "1.0, false",
+            "1, false",
+            "John, false"
+    })    void identifierPatternMatcher(String identifier, boolean expected) {
```

Comment:
Not sure if this is something you want to keep? I'd assume this was tested via a handler?

Reply:
It has done it's job.. removed